packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -72,6 +75,58 @@ export function Comments( {
 		);
 	}, [] );
 
+	const { selectedClientBlocks, selectedActiveClientId } = useSelect(
+		( select ) => {
+			const clientID =
+				select( blockEditorStore ).getSelectedBlockClientId();
+			const selClientBlocks = select( blockEditorStore ).getBlocks();
```

Comment:
Why not run `getBlocks` only when it's needed, inside `handleThreadClick`? This selector will cause the whole component to re-render every time a block changes.

Reply:
@ellatrix - Updated the `getBlocks` to load inside `handleThreadClick`

Reply:
Great, that looks much simpler too :)